web/modules/custom/joinup_core/src/Form/RdfEntityUriWebConverterForm.php

```diff
@@ -56,29 +62,35 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
     }
 
     // Build a Drupal internal URI.
-    $uri = 'internal:/' . trim($path, '/');
+    $uri = 'internal:/' . $path;
     $url = Url::fromUri($uri);
     // The path was build from a route.
     if ($url->isRouted()) {
       // This path contains a RDF parameter.
       $parameters = $url->getRouteParameters();
       if (!empty($parameters['rdf_entity'])) {
         $form_state->set('passed_as', 'url');
-        $form_state->setValue('rdf_entity_id', UriEncoder::decodeUrl($parameters['rdf_entity']));
+        $form_state->setValue('entity_id', UriEncoder::decodeUrl($parameters['rdf_entity']));
+        return;
+      }
+      elseif (!empty($parameters['taxonomy_term'])) {
+        $form_state->set('passed_as', 'url');
+        $form_state->setValue('entity_id', UriEncoder::decodeUrl($parameters['taxonomy_term']));
         return;
       }
     }
-    $form_state->setErrorByName('id', $this->t('The entered value (:id) is neither a valid RDF entity ID, nor a RDF entity URL.', [':id' => $id]));
+    $form_state->setErrorByName('id', $this->t('The entered value (:id) is neither a valid RDF entity ID, nor a valid RDF taxonomy ID, nor a RDF entity URL.', [':id' => $id]));
   }
 
   /**
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
     $passed_as = $form_state->get('passed_as');
-    $id = $form_state->getValue('rdf_entity_id');
+    $id = trim($form_state->getValue('entity_id'), " /");
+    $entity_type = $form_state->get('entity_type');
```

Comment:
It's a good practice to use core naming: `$entity_type_id`. `$entity_type` is the object.